lib/events/dynamoevents/dynamoevents.go

```diff
@@ -1067,14 +1067,28 @@ func (l *Log) getTableStatus(ctx context.Context, tableName string) (tableStatus
 }
 
 func (l *Log) getBillingModeIsProvisioned(ctx context.Context) (bool, error) {
-	table, err := l.svc.DescribeTableWithContext(ctx, &dynamodb.DescribeTableInput{
+	res, err := l.svc.DescribeTableWithContext(ctx, &dynamodb.DescribeTableInput{
 		TableName: aws.String(l.Tablename),
 	})
 	if err != nil {
 		return false, trace.Wrap(err)
 	}
 
-	return *table.Table.BillingModeSummary.BillingMode == dynamodb.BillingModeProvisioned, nil
+	// Guaranteed to be set.
+	table := res.Table
+
+	// Perform pessimistic nil-checks, assume the table is provisioned if they are true.
+	// Otherwise, actually check the billing mode.
+	switch {
```

Comment:
I think we can simplify this switch down to a single boolean expression?

```suggestion
	return table.BillingModeSummary == nil || 
	table.BillingModeSummary.BillingMode == nil 
	|| *table.BillingModeSummary.BillingMode == dynamodb.BillingModeProvisioned, nil
```

Reply:
Yep, seems reasonable.